A/B Test edit pass /1 (#28551)
dotnet/AspNetCore.Docs: aspnetcore/migration/inc/abtesting.md

uid: migration/inc/ab-testing
---

# A/B Testing endpoints during migration

During incremental migration, new endpoints are brought over to a [YARP](https://microsoft.github.io/reverse-proxy/) enabled ASP.NET Core app. With the default setup, these endpoints are automatically served for all requests once deployed. In order to test these endpoints, or be able to turn them off if needed, additional setup is needed.

This document describes how to setup a conditional endpoint selection system to enable A/B testing during incremental migration. It assumes a setup as described in [incremental migration overview](xref:migration/inc/overview) as a starting point.

## Conditional endpoint selection

To enable conditional endpoint selection, a few services need to be defined:

> 1. Metadata that can be added to an endpoint to turn on any conditional related logic. Endpoints include controllers, minimal APIs, Razor Page, etc. If this metadata isn't present on an endpoint, that endpoint won't incur any potential cost with checks that may be performed.

Author: If this metadata isn't present on an endpoint, that endpoint won't incur any potential cost with checks that may be performed.
That's hard to parse. Can it be written simpler?

Reviewer: how about "If this metadata isn't added to an endpoint, no conditional checks will be performed for it."

> 1. Metadata that can be added to an endpoint to turn on any conditional related logic. Endpoints include controllers, minimal APIs, Razor Page, etc. If this metadata isn't present on an endpoint, that endpoint won't incur any potential cost with checks that may be performed.

Author: ```suggestion
1. Metadata that can be added to an endpoint to turn on any conditional related logic. Endpoints include controllers, minimal APIs, Razor Page, etc. If this metadata isn't added to an endpoint, no conditional checks are performed for it.
```
